### PR TITLE
Add 'deny' password authenticator

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
@deads2k

/cc @brenton adds a deny option, per https://trello.com/c/XlDJ66hG/376-secure-out-of-the-box-authentication

File: pkg/auth/authenticator/password/denypassword/denypassword.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/auth/user"
+	"github.com/openshift/origin/pkg/auth/authenticator"
+)
+
+// denyPasswordAuthenticator denies all password requests
+type denyPasswordAuthenticator struct {
+}
+
+// New creates a new password authenticator that approves any login attempt with non-blank username and password
```
Comment (deads2k) [inline pkg/auth/authenticator/password/denypassword/denypassword.go]: Bad comment

Comment (liggitt) [inline pkg/auth/authenticator/password/denypassword/denypassword.go]: fixed

**Comment by deads2k** [comment]
Fix the comment, then merge.

**Comment by liggitt** [comment]
[merge]

**Comment by brenton** [comment]
awesome!

**Pull request merged**